regions/shapes/moc/core.py

```diff
@@ -0,0 +1,941 @@
+        interval_set = IntervalSet.from_numpy_array(intervals_arr)
+        return cls(interval_set, meta, visual)
+
+    @classmethod
+    def from_skycoords(cls, skycoords, max_norder, meta=None, visual=None):
+        """
+        Create a `~regions.MOCSkyRegion` from a `astropy.coordinates.SkyCoord` object.
+
+        Parameters
+        ----------
+        skycoords : `astropy.coordinates.SkyCoord`
+            A set of astropy formatted skycoords.
+        max_norder : int
+            The maximum order that we want for the new `~regions.MOCSkyRegion`.
+        meta : `~regions.RegionMeta` object, optional
+            A dictionary which stores the meta attributes of this region.
+        visual : `~regions.RegionVisual` object, optional
+            A dictionary which stores the visual meta attributes of this region.
+
+        Returns
+        -------
+        moc : `~regions.MOCSkyRegion`
+            A new `~regions.MOCSkyRegion` object created from a `astropy.coordinates.SkyCoord`
+        """
+        hp = HEALPix(nside=(1 << max_norder), order='nested')
+        ipix = hp.lonlat_to_healpix(skycoords.icrs.ra, skycoords.icrs.dec)
+
+        shift = 2 * (MOCSkyRegion.HPY_MAX_NORDER - max_norder)
+        intervals_arr = np.vstack((ipix << shift, (ipix + 1) << shift)).T
+
+        interval_set = IntervalSet.from_numpy_array(intervals_arr)
+        return cls(interval_set, meta, visual)
+
+    @classmethod
+    def from_lonlat(cls, lon, lat, max_norder, meta=None, visual=None):
+        """
+        Create a `~regions.MOCSkyRegion` from ``lon``, ``lat`` `astropy.units.Quantity`.
+
+        Parameters
+        ----------
+        lon : `astropy.units.Quantity`
+            A set of astropy formatted ra quantities.
+        lat : `astropy.units.Quantity`
+            A set of astropy formatted dec quantities.
+        max_norder : int
+            The maximum order that we want for the new `~regions.MOCSkyRegion`.
+        meta : `~regions.RegionMeta` object, optional
+            A dictionary which stores the meta attributes of this region.
+        visual : `~regions.RegionVisual` object, optional
+            A dictionary which stores the visual meta attributes of this region.
+
+        Returns
+        -------
+        moc : `~regions.MOCSkyRegion`
+            A new `~regions.MOCSkyRegion` object created from lon, lat `astropy.units.Quantity`.
+        """
+        hp = HEALPix(nside=(1 << max_norder), order='nested')
+        ipix = hp.lonlat_to_healpix(lon, lat)
+
+        shift = 2 * (MOCSkyRegion.HPY_MAX_NORDER - max_norder)
+        intervals_arr = np.vstack((ipix << shift, (ipix + 1) << shift)).T
+
+        interval_set = IntervalSet.from_numpy_array(intervals_arr)
+        return cls(interval_set, meta, visual)
+
+    @property
+    def max_order(self):
+        """
+        Returns the deepest order needed to describe self's interval set.
+        """
+        # TODO: cache value
```

Comment:
What about this TODO? Please remove or address it.

(we should take this PR as an opportunity to improve and polish, once code has landed often this just doesn't happen any more; so please be a bit patient if I come with nitpicky comments in the coming days)